html2docx/html2docx.py

```diff
@@ -96,11 +97,30 @@ def init_p(self, attrs: List[Tuple[str, Optional[str]]]) -> None:
             elif style_decl["name"] == "padding-left" and style_decl["unit"] == "px":
                 self.padding_left = Pt(style_decl["value"])
 
+    def init_table(self, attrs: List[Tuple[str, Optional[str]]]) -> None:
+        self.table_data: List[List[str]] = []
+
     def finish_p(self) -> None:
         if self.r is not None:
             self.r.text = self.r.text.rstrip()
         self._reset()
 
+    def finish_table(self) -> None:
+        if self.table_data:
```

Comment:
The HTML document must be valid, thus a table close tag can only appear after a table open tag, no need to check if table data has been initialized.
```suggestion
```